typescript/index.ts

```diff
@@ -0,0 +1,50 @@
+import { LogLevel, Player, PlayerConfig, SourceConfig } from 'bitmovin-player/modules/bitmovinplayer-core';
+import PolyfillModule from 'bitmovin-player/modules/bitmovinplayer-polyfill';
+import EngineBitmovinModule from 'bitmovin-player/modules/bitmovinplayer-engine-bitmovin';
+import MseRendererModule from 'bitmovin-player/modules/bitmovinplayer-mserenderer';
+import HlsModule from 'bitmovin-player/modules/bitmovinplayer-hls';
+import XmlModule from 'bitmovin-player/modules/bitmovinplayer-xml';
+import DashModule from 'bitmovin-player/modules/bitmovinplayer-dash';
+import AbrModule from 'bitmovin-player/modules/bitmovinplayer-abr';
+import ContainerTSModule from 'bitmovin-player/modules/bitmovinplayer-container-ts';
+import ContainerMp4Module from 'bitmovin-player/modules/bitmovinplayer-container-mp4';
+import SubtitlesModule from 'bitmovin-player/modules/bitmovinplayer-subtitles';
+import SubtitlesCEA608Module from 'bitmovin-player/modules/bitmovinplayer-subtitles-cea608';
+import StyleModule from 'bitmovin-player/modules/bitmovinplayer-style';
+
+import { UIFactory } from 'bitmovin-player/bitmovinplayer-ui';
```

Comment:
This raises a Typescript error because there is no `.d.ts` file:
```
Could not find a declaration file for module 'bitmovin-player/bitmovinplayer-ui'. '/Users/daniel/Projects/bitmovin-player-web-samples/typescript/node_modules/bitmovin-player/bitmovinplayer-ui.js' implicitly has an 'any' type.
```

In the `modules/` folder there's a `bitmovinplayer-ui.d.ts` file but it only contains this:
```ts
// Dummy default module export as placeholder for the real module
declare const module: any;
export default module;
```
This means changing this line to use `bitmovin-player/modules/bitmovinplayer-ui` only changes the problem as `UIFactory` is not exported according to the types.

Any recommendations @felix-hoc @HalfbyteHeroes @bitmovin-kenny ?

Reply:
I think the dummy declaration file should be fine, since we have one for each module. But for the ones that are actually available, we have overrides in the `types/` directory...

I'm actually not sure if this is expected because we just don't export the UI types - or if there is a problem. Maybe someone else knows.

That being said, you can make it work by adding the UI as its own dependency, and then import the factory from there:
```ts
import { UIFactory } from 'bitmovin-player-ui';
```

Reply:
Good point, didn't think of simply importing it separately. That does the trick, nice - thanks! 

However, wouldn't it make sense to distribute the UI types as well in the player NPM package as we also ship the UI JS/CSS files? 🤔